src/plugin/datasource.ts

```diff
@@ -515,4 +575,51 @@ export function WavefrontDatasource(instanceSettings, $q, backendSrv, templateSr
         return this.backendSrv.datasourceRequest(reqConfig);
     };
 
+
+    function refreshToken() {
+        if (instanceSettings.jsonData.cspAPIToken) {
+            try {
+                fetch(CSP_API_TOKEN_URL, {
+                    method: "POST",
+                    body: JSON.stringify({
+                        "api_token": instanceSettings.jsonData.cspAPIToken,
+                    }),
+                    headers: {
+                        "Content-type": "application/x-www-form-urlencoded; charset=UTF-8"
+                    }
+                })
+                .then((response) => response.json())
+                .then((json) => console.log(json));
+            } catch(e) {
+                console.error(e);
+            }
+            this.requestConfigProto.headers["Authorization"] = "Bearer " + instanceSettings.jsonData.cspAPIToken;
+        } else if (instanceSettings.jsonData.cspOAuthClientId && instanceSettings.jsonData.cspOAuthClientSecret) {
+            try {
+                fetch(CSP_OAUTH_TOKEN_URL, {
+                    method: "POST",
+                    body: JSON.stringify({
+                        "grant_type": "client_credentials",
+                    }),
+                    headers: {
+                        "Content-type": "application/x-www-form-urlencoded; charset=UTF-8",
+                        "Authorization": credentials
+                    }
+                })
+                .then((response) => response.json())
+                .then((json) => console.log(json));
+            } catch(e) {
+                console.error(e);
+            }
+            this.requestConfigProto.headers["Authorization"] = "Bearer " + instanceSettings.jsonData.cspAPIToken;
+        }
+        console.log("Refreshed token!");
+      }
+      
+      // Execute the function initially
+      refreshToken();
```

Comment:
Since, we are already calling it during initialisation at line#33, we can skip this initial call of refreshToken